redis/src/cluster_client.rs

```diff
@@ -82,8 +82,8 @@ pub(crate) struct ClusterParams {
     pub(crate) tls: Option<TlsMode>,
     pub(crate) retry_params: RetryParams,
     pub(crate) tls_params: Option<TlsConnParams>,
-    pub(crate) connection_timeout: Duration,
-    pub(crate) response_timeout: Duration,
+    pub(crate) connection_timeout: Option<Duration>,
```

Comment:
Sorry, I vaguely recall we discussed this elsewhere, but remind me why we're doing this? Can we fix the warnings some other way? It seems to me like things are more clear if the defaults are defined here rather than in the connect method.

Reply:
this is done in order to bridge the gap between sync & async clients. ATM the sync client doesn't use these fields, so if we want to use these fields in the sync clients, either we introduce new defaults to the sync client or change the fields to `Option` and set the defaults in the async client.

Reply:
We could also just potentially add an `allow_unused` that's specific to  `!cluster_async` feature?

Reply:
we could, but wouldn't it be better if these fields actually did what they say they do?

Reply:
Yep, I agree. I think adding defaults for the sync client is a good idea.

Reply:
just making sure - you want to apply the same defaults we currently have on the async client to the sync client?

Reply:
Yes, albeit in a separate PR. Alternatively, I'm ok with just allowing the fields to be unused, which although goofy, is effectively what we currently have, right?

Reply:
right.
https://github.com/redis-rs/redis-rs/pull/1333